js/utils/Socket.js

```diff
@@ -0,0 +1,64 @@
+var __ = require('underscore'),
```

Comment:
This is a wrapper class for Sockets which I created since the native Socket interface is quite bare-bones. Some of the cool things it offers.

- It extends the Backbone Event object so you have a pretty robust event api. The native implementation is to set a function on a property, like:
`socket.onconnect = function() {}`

That's not so friendly if different pieces of code want to bind events. One person would be blowing away the other person's handler, or they would need to wrap it. Now we could use all of BB's event methods, like on, off, trigger... e.g.:

`socket.on('close', function(){});`

in some other module where you have a reference to the same socket:
`socket.on('close', function(){});`

- the other cool thing that it adds is... there is no way with native sockets to reconnect upon loss of a connect. You're just supposed to instantiate a new socket. The problem with that is any event handlers have to be rebound to the new instance. This wrapper class abstracts that away from you, so on re-connect all your handlers will still be bound. In actuality, they're bound to the wrapper not the instance and the wrapper proxies the instances events to you.

I haven't refactored the existing socket view to use this. If we like that's something we could do. I did use this wrapper for the heartbeat socket.